source/lac/trilinos_sparse_matrix.cc

```diff
@@ -2310,6 +2310,15 @@ namespace TrilinosWrappers
     return ((sizeof(TrilinosScalar)+sizeof(TrilinosWrappers::types::int_type))*
             matrix->NumMyNonzeros() + sizeof(int)*local_size() + static_memory);
   }
+
+  MPI_Comm SparseMatrix::get_mpi_communicator () const
+  {
+    const Epetra_MpiComm *comm = dynamic_cast<const Epetra_MpiComm *>(&range_partitioner().Comm());
+    if (comm)
+      return comm->Comm();
+    else
+      return MPI_COMM_WORLD;
+  }
```

Comment:
Did you check this code on a Trilinos without MPI? I'm wondering whether Trilinos actually provides Epetra_MpiComm in case it is compiled without MPI. Have a look at include/deal.II/lac/trilinos_vector_base.h around line 1990 for how we do this for the vector. (In general, I like adding the MPI communicator here.)